### PR TITLE
Add instructor checklist to exam details page

### PR DESCRIPTION
**Pull request by nicolasruscher**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
Implement request https://github.com/ls1intum/Artemis/issues/2718

### Description
We add multiple usability improvements. 
1. Exam Titles within the exam management page work as links to the exam details page. 
2. For each exam within the exam management page, there is a checklist button that links to the checklist within the exam detail page
3. The exam detail page contains a new table 'checklist' with contains general information about how to conduct an exam with Artemis, as well as direct links to the aforementioned pages
4. Buttons to all exam pages have been added to the exam detail page

### Steps for Testing
Make sure that all new buttons link to the correct pages and that all phrases are correctly displayed in German and English.

### Screenshots

Exam Management Page:
Bildschirmfoto 2021-02-02 um 13 26 18
Exam Detail Page:
Bildschirmfoto 2021-02-02 um 16 03 55

### CODE REVIEW

File: src/main/webapp/i18n/en/exam.json [hunk 1]
```diff
@@ -297,6 +298,42 @@
...
+                    "testRun": "Conduct a test run",
+                    "exportResults": "Export the exam results"
+                },
+                "descriptionItem": {
+                    "register": "At least one student has to be registered to the exam",
+                    "exerciseGroups": "The Points in all exercises within a given exercise group must be identical, and the number of exercise groups must equal the exams number of exercises.",
+                    "generateStudentExams": "Student exams must be generated for all registered students before starting the exam.",
+                    "prepareExerciseStart": "Preparing the exercises can reduce load on the server during the exam.",
+                    "examDetails": "Before conducting an exam, please make sure to have filled out all exam details. (Start text, dates etc)",
+                    "conductExam": "On the student exam page it can be seen with students did already submit their exam.",
```
Comment (JohannesWeiss) [inline src/main/webapp/i18n/en/exam.json:329]: With → Which

File: src/main/webapp/i18n/de/exam.json [hunk 2]
```diff
@@ -296,6 +297,42 @@
             "delete": {
                 "question": "Soll die Klausur <strong>{{ title }}</strong> wirklich dauerhaft gelöscht werden? Alle zugehörigen Elemente werden gelöscht, inklusive der Klausuren der Studenten. Diese Aktion kann NICHT rückgängig gemacht werden!",
                 "typeNameToConfirm": "Bitte gib den Namen der Klausur zur Bestätigung ein."
+            },
+            "checklist": {
+                "title": "Checkliste",
+                "introduction": "Unter folgendem Link können sie mehr über den Ablauf einer Artemis Klausur informieren:",
```
Comment (JohannesWeiss) [inline src/main/webapp/i18n/de/exam.json:303]: sie → Sie sich

File: src/main/webapp/i18n/de/exam.json [hunk 3]
```diff
@@ -296,6 +297,42 @@
...
+                    "exportResults": "Exportiere die Klausurergebnisse"
+                },
+                "descriptionItem": {
+                    "register": "Mindestens ein Studierender muss zur Klausur angemeldet sein.",
+                    "exerciseGroups": "Die Punkte aller Aufgaben einer Aufgabengruppe muss identisch sein. Die Anzahl an Aufgabengruppen muss der Anzahl der Aufgaben in der Klausur entsprechen.",
+                    "generateStudentExams": "Die Klausuren der Studierenden müssen generiert worden sein, bevor die Klausur beginnt.",
+                    "prepareExerciseStart": "Vorbereiten der Aufgaben kann die Serverlast während der Klausur reduzieren.",
+                    "examDetails": "Alle relevanten Details der Klausur müssen vor dem Start der Korrektur eingetragen sein.",
+                    "conductExam": "In der Studierenden-Klausurübersicht sieht man, welche Studierende ihre Klausur bereits abgegeben haben.",
+                    "assessAllSubmissions": "Die Anzahl der Korrekturrunden der Aufgaben entspricht der in der Klausur angegebenen Anzahl von Korrekturrunden.",
```
Comment (JohannesWeiss) [inline src/main/webapp/i18n/de/exam.json:329]: Korrekturrunden → Korrektur-Runden
Improves readability

**Review by airelawaleria — COMMENTED** [review]
Everything worked as expected :)

Only found something small. When zooming in, the alignment of the buttons/padding doesn't look nice (on the screenshot the zoom is 100% in chrome)
image

**Review by JohannesWeiss — COMMENTED** [review]
Cool new feature, works as expected. 
Just a few typos remain.

File: src/main/webapp/i18n/de/exam.json (hunk 2 above)
Comment (krusche) [reply src/main/webapp/i18n/de/exam.json:303]: please always use "du" instead of "Sie"

File: src/main/webapp/i18n/de/exam.json (hunk 3 above)
Comment (krusche) [reply src/main/webapp/i18n/de/exam.json:329]: to be honest, I am not a big fan of those dashes (`-`) between words. I would like to get rid of all of those ;-)

Comment (nicolasruscher) [reply src/main/webapp/i18n/de/exam.json]: Done

File: src/main/webapp/i18n/de/exam.json (hunk 2 above)
Comment (nicolasruscher) [reply src/main/webapp/i18n/de/exam.json]: Done

File: src/main/webapp/i18n/en/exam.json (hunk 1 above)
Comment (nicolasruscher) [reply src/main/webapp/i18n/en/exam.json]: Done

**Review by tiffca — DISMISSED** [review]
Everything works as expected 

Some minor typos: 
"On the student exam page it can be seen which students did already submit their exam" -> "On the student exam page it can be seen which students already **submitted** their exam" [changed wording]

"The number of correction rounds per exercise depends on the **exams** number of correction rounds." -> "The number of correction rounds per exercise depends on the **exam's** number of correction rounds." [added apostrophe in exams]

"and the number of exercise groups must equal the **exams** number of exercises." -> "and the number of exercise groups must equal the **exam's** number of exercises." [added apostrophe in exams]

**Review by ivanchimeno — DISMISSED** [review]
I'm approving because the code looks fine for me :)

**Review by SimonEntholzer — DISMISSED** [review]
Code looks good, everything was shonw as expected while testing :D

**Review by philgabr — COMMENTED** [review]
All buttons work as expected.

Seems like there is one translation missing:
(Also I'd agree with @airelawaleria, that paddings/margins could be improved)
image

**Review by FlorianStang — COMMENTED** [review]
Same as @philgabr

File: src/test/javascript/spec/component/exam/manage/exams/exam-detail.component.spec.ts [hunk 4]
```diff
@@ -56,17 +57,20 @@ describe('ExamDetailComponent', () => {
...
                     provide: ActivatedRoute,
                     useValue: {
+                        snapshot: {},
                         data: {
                             subscribe: (fn: (value: Data) => void) =>
                                 fn({
                                     exam,
                                 }),
                         },
+                        snapshot: {},
```
Comment (fde312) [inline src/test/javascript/spec/component/exam/manage/exams/exam-detail.component.spec.ts:73]:
```suggestion

```

**Review by fde312 — APPROVED** [review]
Code looks good and works fine! Good job! 👍

**Review by SimonEntholzer — APPROVED** [review]
We tried to fix the test, it's fixed now. Everyhting else works fine too :D

**Pull request merged**